client-permissions/client.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/sh
+#
+# Environment variables used:
+#  - SERVER: hostname or IP-address of the NFS-server
+#  - EXPORT: NFS-export to test (should start with "/")
+
+# enable some more output
+set -x
+
+[ -n "${SERVER}" ]
+[ -n "${EXPORT}" ]
+
+if [ "$1" = "client_initialization" ]
+then
+	# install build and runtime dependencies
+	yum -y install nfs-utils time
+	mkdir -p /mnt/ganesha
+	
+	echo "------------------------------------------------------------------------"
+	echo "Client Initial Stage --- With All Rights To All Clients ( RO & RW ) "
+	echo "------------------------------------------------------------------------"
+	
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+	cd /mnt/ganesha
+	echo "Trying To Write A File"
+	echo "Hello World" > testFile.txt
+	ret=$?
+	if [ $ret -ne 0 ]
+	then
+		echo "FAILURE: Write permissions denied"
+		exit ret
+	fi
+	echo "Trying To Read A File"
+	cat testFile.txt
+	ret=$?
+	if [ $ret -ne 0 ]
+	then
+		echo "FAILURE: Read permissions denied"
+		exit ret
+	fi
+	echo "SUCCESS: With all rights to all Clients ( RO & RW )"
+	#unmount
+	cd / && umount -l /mnt/ganesha
+fi
+
+if [ "$1" = "client_stage1" ]
+then
+	echo "------------------------------------------------------------------------"
+	echo "Client Stage 1 --- With Only RO Rights To This Client "
+	echo "------------------------------------------------------------------------"
+
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+	cd /mnt/ganesha
+	echo "Trying To Write A File"
+	sed -i '1s/$/ From RedHat/' testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "FAILURE: Write permissions were not blocked to the Client"
+		exit ret
+	fi
+	echo "Trying To Read A File"
+	cat testFile.txt
+	ret=$?
+	if [ $ret -ne 0 ]
+	then
+		echo "FAILURE: Read permissions denied"
+		exit ret
+	fi
+	echo "SUCCESS: With Only RO Rights To This Client"
+	# unmount
+	cd / && umount -l /mnt/ganesha
+fi
+
+
+if [ "$1" = "client_stage2" ]
+then
+	echo "------------------------------------------------------------------------"
+	echo "Client Stage 2 --- With Only Rights For v3 Mount To This Client "
+	echo "------------------------------------------------------------------------"
+
+	echo "Trying To Mount By vers=3"
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+	ret=$?
+	if [ $ret -ne 0 ]
+	then
+		echo "FAILURE: Mount v3 failed"
+		exit ret
+	else
+		#unmount version 3 
+		cd / && umount -l /mnt/ganesha
+	fi
+	
+	echo "Trying To Mount By vers=4.0"
+	mount -t nfs -o vers=4.0 ${SERVER}:${EXPORT} /mnt/ganesha
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "FAILURE: Mount v4.0 Permissions were not blocked to the Client"
+		exit ret
```

Comment:
You probably do not want to return `$ret` here, it is `0` and that cause the failure in the test to exit the script with success.

Reply:
@nixpanic Done